Nos/Views/NoteComposer/NoteComposer.swift

```diff
@@ -185,6 +192,63 @@ struct NoteComposer: View {
         .alert(unwrapping: $alert)
     }
 
+    /// Action Bar displayed below the editing controller.
+    private var composerActionBar: some View {
+        ComposerActionBar(
+            editingController: $editingController,
+            expirationTime: $expirationTime,
+            isUploadingImage: $isUploadingImage,
+            showPreview: $showNotePreview
+        )
+    }
+
+    /// Note Preview displayed above the editing controller when the Preview switch is turned on.
+    private var notePreview: some View {
+        VStack(spacing: 0) {
+            Group {
+                if let previewEvent {
+                    ScrollView {
+                        NoteButton(
+                            note: previewEvent,
+                            shouldTruncate: false,
+                            hideOutOfNetwork: false,
+                            repliesDisplayType: .displayNothing,
+                            fetchReplies: false,
+                            displayRootMessage: true,
+                            isTapEnabled: false,
+                            replyAction: nil,
+                            tapAction: nil
+                        )
+                        .id(previewEvent)
```

Comment:
This interacts with .disabled(showNotePreview). Without the line that disables the note composer, the user can still edit the note while displaying the preview. With that line, we prevent the user from editing while previewing. However, while in macOS that works perfectly, in iOS the preview never loads, it shows the preview briefly and then an infinite spinner wheel. I don't know why exactly that happens. I tried different approaches, like trying with resignFirstResponder() in UIKit, but nothing worked, so I ended adding this line here that prevents SwiftUI from rebuilding what's inside. I didn't find anything wrong with it, not a regression, but it is definitely something we need to carefully inspect for side changes.

Reply:
Weird, but I don't have any better ideas. I mean we could tear down the NoteTextEditor when showing the preview and then re-create it. But it sounds like an approach that would likely present its own problems. I'm ok with the `.id()` modifier for situations like this.

Reply:
Same. I don't have any better ideas, either. Nice job figuring this all out.